### ISSUE TITLE
Update NPM packages duplicate devDependencies to dependencies in package.json

### ISSUE THREAD

**Issue by TonioOoOo** [Microsoft/nodejstools#1235]
##### Expected Behavior

Dev dependencies stay in `devDependencies` in `package.json` with the new version number when i update npm packages via context menu in solution explorer.
##### Actual Behavior

When i use "Update npm Packages" in npm context menu in Solution explorer :
Updated  `devDependencies` move in `dependencies` zone in package.json (with the good version number) and stay in  `devDependencies` with the old version number.

![devdp](https://cloud.githubusercontent.com/assets/9881996/18058486/e8221606-6e14-11e6-8845-c92ad967cf7c.png)

- NTVS Version: 1.2.40726.00
- Visual Studio Version: 14.0.25425.01 Update 3
- Node.js Version: 6.4.0
##### Steps to Reproduce
1. Install a package dependency type "Development" in an old version and add in package.json ;
2. Update via context menu in solution explorer ;
3. Open package.json ;

**Comment by mjbvz** [comment]
Thanks for the report. I confirmed this.

This is not NTVS specific, but also a weird npm behavior for the update command:

```
> npm install --save-dev express@3.0.x
> type package.json
{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "express": "^3.0.6"
  }
}
>npm update --save
>type package.json
{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "express": "^3.0.6"
  },
  "dependencies": {
    "express": "^3.21.2"
  }
}
```

I'll investigate to confirm this is actually expected and if there are any good workarounds we can do on our end.

**Comment by mjbvz** [comment]
This certainly seems like a bug on the npm side. I've tried messing around with the `--dev`, `--save`, and `--save-dev` flags in various combinations and the following is all I've found so far that seems to work properly when dealing with both dev and non-dev dependencies:

```bash
> npm update --save-dev & npm update --save
```

**Comment by mjbvz** [comment]
Ahh, here's the npm bug tracking this: https://github.com/npm/npm/issues/11622

**Comment by TonioOoOo** [comment]
Ah ok, sorry for not taking the time to look further. I close the ticket? Or we expect the closing of the NPM ticket?

**Comment by mjbvz** [comment]
No problem. 

This bug is an interesting case. The details of how we update the npm packages should really be hidden to the user and I think there are workarounds we could look into (such as running `npm update --save-dev & npm update --save`). However given the possibility of those workarounds breaking other scenarios, I'm more inclined to say we should get the fix in npm instead (similar to #89).

Let's keep this bug open to make sure we follow up on this.

Thanks.

**Comment by paulvanbrenk** [comment]
I no longer reproduce this with npm 5.5.1
